File: 01000/BOJ_1205 등수 구하기.java
// https://www.acmicpc.net/problem/1205
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        LinkedList<Integer> rankList = new LinkedList<>();
        
        int n = Integer.parseInt(st.nextToken());
        int score = Integer.parseInt(st.nextToken());
        int p = Integer.parseInt(st.nextToken());

        if (n > 0) {
            Arrays.stream(br.readLine().split(" "))
                .mapToInt(Integer::parseInt)
                .forEach(rankList::add);
        }

        if(n == p && score <= rankList.getLast()){
            System.out.println(-1);
        } else {
            long result = rankList.stream()
                .filter(x -> x > score)
                .count() + 1;

            System.out.println(result);
        }
    }
}

File: 01000/BOJ_1205 등수 구하기.c
// https://www.acmicpc.net/problem/1205
#include <stdio.h>

int n, score, p;
int idx = 1, same;

int main()
{
    scanf("%d %d %d", &n, &score, &p);

    while(n--) {
        int k;
        scanf("%d", &k);

        if(k > score) idx++;
        else if(k == score) same++;
    }

    printf("%d", idx + same <= p ? idx : -1);
    return 0;
}